components/post_view/acknowledgements/post_acknowledgements.tsx

```diff
@@ -0,0 +1,147 @@
+
+import PostAcknowledgementsUserPopover from './post_acknowledgements_users_popover';
+
+import './post_acknowledgements.scss';
+
+type Props = {
+    acknowledgedAt: number;
+    currentUserId: UserProfile['id'];
+    hasReactions: boolean;
+    list?: Array<{user: UserProfile; acknowledgedAt: PostAcknowledgement['acknowledged_at']}>;
+    postId: Post['id'];
+}
+
+function moreThan5minAgo(time: number) {
+    const now = new Date().getTime();
+    return now - time > 5 * 60 * 1000;
+}
+
+function PostAcknowledgements({
+    acknowledgedAt,
+    currentUserId,
+    hasReactions,
+    list,
+    postId,
+}: Props) {
+    const dispatch = useDispatch();
+    const [open, setOpen] = useState(false);
+
+    const {x, y, reference, floating, strategy, context} = useFloating({
+        open,
+        onOpenChange: setOpen,
+        placement: 'top-start',
+        whileElementsMounted: autoUpdate,
+        middleware: [
+            offset(5),
+            flip({fallbackPlacements: ['bottom', 'right'], padding: 5}),
+            shift(),
+        ],
+    });
+
+    const headingId = useId();
+
+    const {getReferenceProps, getFloatingProps} = useInteractions([
+        useHover(context, {
+            enabled: list && list.length > 0,
+            delay: {
+                open: 300,
+                close: 0,
+            },
+            handleClose: safePolygon(),
+        }),
+        useRole(context),
+    ]);
+
+    const handleClick = () => {
+        if (acknowledgedAt) {
+            dispatch(unacknowledgePost(postId));
+        } else {
+            dispatch(acknowledgePost(postId));
+        }
+    };
+
+    const button = (
+        <button
+            ref={reference}
+            onClick={handleClick}
+            disabled={Boolean(acknowledgedAt) && moreThan5minAgo(acknowledgedAt)}
+            className={classNames({
+                AcknowledgementButton: true,
+                'AcknowledgementButton--acked': Boolean(acknowledgedAt),
+            })}
+            {...getReferenceProps()}
+        >
+            <CheckCircleOutlineIcon size={16}/>
+            {(list && list.length > 0) ? list!.length : (
+                <FormattedMessage
+                    id={'post_priority.button.acknowledge'}
+                    defaultMessage={'Acknowledge'}
+                />
+            )}
+        </button>
+    );
+
+    if (!list || !list.length) {
+        return button;
+    }
+
+    return (
+        <>
+            {button}
+            {open && (
+                <FloatingFocusManager
+                    context={context}
+                    modal={false}
+                >
+                    <div
+                        ref={floating}
+                        style={{
+                            position: strategy,
+                            top: y ?? 0,
+                            left: x ?? 0,
+                            width: 248,
+                            zIndex: 999,
```

Comment:
Does this mean that this is going to be in the DOM and not something using a portal? I wouldn't have expected that we'd need to set a z-index with floating-dom

Reply:
It's not a portal no... BTW the link about z-index on floating-ui https://floating-ui.com/docs/misc#z-index-stacking

Reply:
Interesting. That's one of the things I would have assumed would be done automatically by the library. I guess we'll have to keep that in mind when unifying all the tooltip stuff eventually